protocol/eip-7702-aliasing.md

```diff
@@ -0,0 +1,78 @@
+# Address Aliasing for EIP-7702
+
+## Purpose
+
+Describes and unifies an updated logic for address aliasing that properly accounts for the Pectra
+hardfork and the inclusion of [EIP-7702][1].
+
+## Problem Statement
+
+[EIP-7702][1] introduces a new mechanism that allows EOAs to specify a delegation that causes the
+EOA to operate as a pseudo-contract that uses the bytecode of the delegation target. A signature
+from the private key can revoke the delegation immediately and at any time. As a result, EOAs that
+have code as a result of EIP-7702 are still functionally EOAs in the sense that they are ultimately
+fully controlled by the account's private key.
+
+The `OptimismPortal` will alias the address in the forced transaction that it generates if the
+sender is not an EOA. We do this check because we do not allow smart contracts on L1 to act as the
+same contract on L2. Contracts on different chains are fundamentally different objects and should
+not be treated as the same even though they have the same address.
+
+Given the change introduced by EIP-7702, we need to figure out how to properly handle EIP-7702
+delegated accounts.
+
+## Proposed Solution
+
+Our existing code would produce the following scenario:
+
+1. `EOA #1` triggers `OptimismPortal`
+2. `OptimismPortal` generates forced tx and does not alias the address of `EOA #1`
+3. `EOA #1` delegates via EIP-7702
+4. `EOA #2` triggers `EOA #3` (possible because of step 3) which triggers `OptimismPortal`
+5. `OptimismPortal` generates forced tx and DOES alias the address of `EOA #1`
```

Comment:
typo? if not, I don't follow this
```suggestion
3. `EOA #1` delegates via EIP-7702
4. `EOA #2` triggers `EOA #1` (possible because of step 3) which triggers `OptimismPortal`
5. `OptimismPortal` generates forced tx and DOES alias the address of `EOA #1`
```